### ISSUE TITLE
FeatureRequest: Specify kinesis stream as the source of a aws_kinesis_firehose_delivery_stream 

### ISSUE THREAD

**Issue by pleschev** [terraform-providers/terraform-provider-aws#1601]

**Comment by pleschev** [comment]
I'm going to have a crack at creating a pull request for this functionality

**Comment by bowenli86** [comment]
Looking forward to this!

**Comment by erikpaasonen** [comment]
Please also include an additional `aws_kinesis_firehose_delivery_stream` output for the modified Firehose Delivery ARN. Currently only the ARN of the Firehose itself is output. But when a Firehose ingests from a Kinesis stream source, there is a separate ARN that is built off of the Firehose AWS service but uses the Kinesis source stream's name, e.g.
`arn:aws:firehose:us-east-1:123456789012:deliverystream/KINESIS_STREAM_NAME`

This additional ARN is required to "reingest" records which failed to process when a Lambda function is used to process the records in Firehose. If you take the standard `aws_kinesis_firehose_delivery_stream` ARN and interpolate it into an IAM policy for the Lambda function like this statement:
``` hcl
    {
      "Sid": "ReingestRecordsToKinesisStream",
      "Effect": "Allow",
      "Action": [
        "firehose:DescribeDeliveryStream",
        "firehose:PutRecord",
        "firehose:PutRecordBatch"
      ],
      "Resource": [
        "${aws_kinesis_firehose_delivery_stream.KINESIS_FIREHOSE_DELIVERY_STREAM_NAME.arn}"
      ]
    },
```
It yields the following error message:

> An error occurred (AccessDeniedException) when calling the PutRecordBatch operation: User: arn:aws:sts::123456789012:assumed-role/ROLE_NAME/SESSION_NAME is not authorized to perform: firehose:PutRecordBatch on resource: arn:aws:firehose:us-east-1:123456789012:deliverystream/KINESIS_STREAM_NAME

**Comment by ewbankkit** [comment]
@bflad This issue was resolved with https://github.com/terraform-providers/terraform-provider-aws/pull/1605 and can be closed.

**Comment by bflad** [comment]
Indeed! This was released a very long time ago in version 1.1.0 of the Terraform AWS Provider.
